src/coreclr/interpreter/compiler.cpp

```diff
@@ -3012,6 +3077,37 @@ bool InterpCompiler::EmitNamedIntrinsicCall(NamedIntrinsic ni, bool nonVirtualCa
 
     switch (ni)
     {
+        case NI_System_Runtime_CompilerServices_AsyncHelpers_Await:
+            // AsyncHelpers.Await should never be expanded here. It is simply needed for recognition elsewhere.
+            return false;
+        case NI_System_Threading_Tasks_Task_ConfigureAwait:
+            // ConfigureAwait should never be expanded here. It is simply needed for recognition elsewhere.
+            return false;
+        case NI_System_Runtime_CompilerServices_AsyncHelpers_AsyncSuspend:
+            if (!m_methodInfo->args.isAsyncCall())
+            {
+                BADCODE("AsyncSuspend should only be used in async methods");
+            }
+            if (m_methodInfo->args.retType != CORINFO_TYPE_VOID)
+            {
+                BADCODE("AsyncSuspend should only be used in async methods with void return type");
+            }
```

Comment:
This condition seems unnecessary.
In the future we may optimize instantiating/unboxing stubs by directly suspending with the callee's continuation, in which case this won't be true.